### PR TITLE
More tests on `Head<T>`, `Tail<T>` and `HasTail<T>` 

### PR DESCRIPTION
**Pull request by balanza**
Additional tests for https://github.com/pagopa/io-ts-commons/pull/370

### CODE REVIEW

**Comment by balanza** [comment]
Unfortunately, test pipeline doesn't start when the base branch is not master. Anyway, the provided test is failing.

I keep the PR on draft until we fix it

File: src/__tests__/types.test.ts [hunk 1]
```diff
@@ -214,3 +214,13 @@ describe("withDefault (composed partial)", () => {
     }
   });
 });
+
+describe(`Head<T>`, () => {
+  it("should map correct type for an array", () => {
+    const c = [1, "two", false];
```
Comment (gquadrati) [inline src/__tests__/types.test.ts:220]: Add `as const` to make it be treated as a struct and not and array.

```suggestion
    const c = [1, "two", false] as const;
```

Comment (balanza) [reply src/__tests__/types.test.ts]: Brilliant! Thanks

File: src/types.ts [hunk 2]
```diff
@@ -278,15 +278,7 @@ export type PromiseType<T> = T extends Promise<infer A> ? A : never;
...
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type Head<T extends ReadonlyArray<any>> = T extends readonly [
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  any,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  ...ReadonlyArray<any>
-]
-  ? T[0]
-  : never;
+export type Head<T extends ReadonlyArray<unknown>> = T[0];
```
Comment (balanza) [inline src/types.ts:281]: @gquadrati

**Review by gquadrati — APPROVED** [review]
Amazing, thank you!

**Pull request merged**